standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/PartitionFilter.g4

```diff
@@ -47,8 +51,13 @@ comparisonOperator
 
 identifier
     : IDENTIFIER #unquotedIdentifer
+    | QUOTEDIDENTIFIER #quotedIdentifier
     ;
 
+QUOTEDIDENTIFIER
+    :
+    ('`'  ( '``' | ~('`') )* '`') { setText(StringUtils.replace(getText().substring(1, getText().length() -1 ), "``", "`")); }
```

Comment:
Could you please move the Java part to `PartFilterVisitor.java` by overriding the method `visitQuotedIdentifier(PartitionFilterParser.QuotedIdentifierContext ctx)`?